var/spack/repos/builtin/packages/vasp/package.py

```diff
@@ -0,0 +1,139 @@
+        # This bunch of 'filter_file()' is to make these options settable
+        # as environment variables
+        filter_file('^CPP_OPTIONS[ ]{0,}=[ ]{0,}',
+                    'CPP_OPTIONS ?= ',
+                    'makefile.include')
+
+        filter_file('^LIBDIR[ ]{0,}=.*$', '', 'makefile.include')
+        filter_file('^BLAS[ ]{0,}=.*$', 'BLAS ?=', 'makefile.include')
+        filter_file('^LAPACK[ ]{0,}=.*$', 'LAPACK ?=', 'makefile.include')
+        filter_file('^FFTW[ ]{0,}?=.*$', 'FFTW ?=', 'makefile.include')
+        filter_file('^MPI_INC[ ]{0,}=.*$', 'MPI_INC ?=', 'makefile.include')
+        filter_file('-DscaLAPACK.*$\n', '', 'makefile.include')
+        filter_file('^SCALAPACK*$', '', 'makefile.include')
+
+        if '+cuda' in spec:
+            filter_file('^OBJECTS_GPU[ ]{0,}=.*$',
+                        'OBJECTS_GPU ?=',
+                        'makefile.include')
+
+            filter_file('^CPP_GPU[ ]{0,}=.*$',
+                        'CPP_GPU ?=',
+                        'makefile.include')
+
+            filter_file('^CFLAGS[ ]{0,}=.*$',
+                        'CFLAGS ?=',
+                        'makefile.include')
+
+        if '+vaspsol' in spec:
+            copy('VASPsol/src/solvation.F', 'src/')
+
+    def setup_build_environment(self, spack_env):
+        spec = self.spec
+
+        cpp_options = ['-DHOST=\\"LinuxGNU\\"', '-DMPI -DMPI_BLOCK=8000',
+                       '-Duse_collective', '-DCACHE_SIZE=4000',
+                       '-Davoidalloc', '-Duse_bse_te',
+                       '-Dtbdyn', '-Duse_shmem']
+
+        cflags = ['-fPIC', '-DADD_']
+
+        spack_env.set('BLAS', spec['blas'].libs.ld_flags)
+        spack_env.set('LAPACK', spec['lapack'].libs.ld_flags)
+        spack_env.set('FFTW', spec['fftw'].prefix)
+        spack_env.set('MPI_INC', spec['mpi'].prefix.include)
+
+        if '+scalapack' in spec:
+            cpp_options.append('-DscaLAPACK')
+            spack_env.set('SCALAPACK', spec['netlib-scalapack'].libs.ld_flags)
+
+        if '+cuda' in spec:
+            cpp_gpu = ['-DCUDA_GPU', '-DRPROMU_CPROJ_OVERLAP',
+                       '-DCUFFT_MIN=28', '-DUSE_PINNED_MEMORY']
+
+            objects_gpu = ['fftmpiw.o', 'fftmpi_map.o', 'fft3dlib.o',
+                           'fftw3d_gpu.o', 'fftmpiw_gpu.o']
+
+            cflags.extend(['-DGPUSHMEM=300', '-DHAVE_CUBLAS'])
+
+            spack_env.set('CUDA_ROOT', spec['cuda'].prefix)
+            spack_env.set('CPP_GPU', ' '.join(cpp_gpu))
+            spack_env.set('OBJECTS_GPU', ' '.join(objects_gpu))
+
+        if '+vaspsol' in spec:
+            cpp_options.append('-Dsol_compat')
+
+        # Finally
+        spack_env.set('CPP_OPTIONS', ' '.join(cpp_options))
+        spack_env.set('CFLAGS', ' '.join(cflags))
+
+    def build(self, spec, prefix):
+        if '+cuda' in self.spec:
+            make('gpu', 'gpu_ncl')
+        else:
+            make()
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
```

Comment:
This line shouldn't be needed, `install_tree` will create a directory if one does not already exist.

Reply:
Fixed on b54750b